CMake/Dependencies/libkvsCommonLws-CMakeLists.txt

```diff
@@ -4,21 +4,43 @@ project(libkvsCommonLws-download NONE)
 
 include(ExternalProject)
 
-ExternalProject_Add(libkvsCommonLws-download
-    GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
-    GIT_TAG           develop
-    PREFIX            ${CMAKE_CURRENT_BINARY_DIR}/build
-    CMAKE_ARGS        
-      -DCMAKE_INSTALL_PREFIX=${OPEN_SRC_INSTALL_PREFIX}
-      -DBUILD_COMMON_LWS=ON
-      -DBUILD_COMMON_CURL=OFF
-      -DBUILD_DEPENDENCIES=FALSE
-      -DOPEN_SRC_INSTALL_PREFIX=${OPEN_SRC_INSTALL_PREFIX}
-      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-      -DUSE_OPENSSL=${USE_OPENSSL}
-      -DUSE_MBEDTLS=${USE_MBEDTLS}
-      -DKVS_DEFAULT_STACK_SIZE=${KVS_DEFAULT_STACK_SIZE}
-      -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
-    BUILD_ALWAYS      TRUE
-    TEST_COMMAND      ""
-)
+if(WIN32)
+  ExternalProject_Add(libkvsCommonLws-download
+      GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
+      GIT_TAG           develop
+      PREFIX            ${CMAKE_CURRENT_BINARY_DIR}/build
+      CMAKE_ARGS        
+        -DCMAKE_INSTALL_PREFIX=${OPEN_SRC_INSTALL_PREFIX}
+        -DBUILD_COMMON_LWS=ON
+        -DBUILD_COMMON_CURL=OFF
+        -DBUILD_DEPENDENCIES=FALSE
+        -DOPEN_SRC_INSTALL_PREFIX=${OPEN_SRC_INSTALL_PREFIX}
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DUSE_OPENSSL=${USE_OPENSSL}
+        -DUSE_MBEDTLS=${USE_MBEDTLS}
+        -DKVS_DEFAULT_STACK_SIZE=${KVS_DEFAULT_STACK_SIZE}
+        -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
+        -DPKG_CONFIG_EXECUTABLE=${PKG_CONFIG_EXECUTABLE}
```

Comment:
```
-DPKG_CONFIG_EXECUTABLE=${PKG_CONFIG_EXECUTABLE}
```

Why is this necessary now but wasn't before?  I am not in favor of having this type of if/else it takes some effort to see what the difference between the two is.  We should find a way to consolidate this so we can either set a single variable that is different before the ExternalProject_Add, or export this variable PKG_CONFIG_EXECUTABLE and make a cmake global so it's available inside producer c build so we don't have to manually specify it.  This variable is set by when the system is trying to find package config.  This seems like instead there is a bug in the windows setup for producer c if it is not able to find package config on its own and needs it to be passed in, then we should fix it there if that is the case instead of duplicate this here.

Reply:
I did this because we already do it for the Windows build in WebRTC. The error said it was specifically missing this path. I'm not sure what changed on Windows to require it now though. I'm using Divya's package manager, hoping to not do this now.